scylla-cdc-replicator/src/replication_tests.rs

```diff
@@ -1296,4 +1296,49 @@ mod tests {
             .await
             .unwrap();
     }
+
+    #[tokio::test]
+    async fn test_range_delete() {
+        let schema = TestTableSchema {
+            name: "RANGE_DELETE".to_string(),
+            partition_key: vec![("pk1", "int"), ("pk2", "int")],
+            clustering_key: vec![("ck1", "int"), ("ck2", "int"), ("ck3", "int")],
+            other_columns: vec![("v", "int")],
+        };
+
+        let operations = vec![
+            (
+                "INSERT INTO RANGE_DELETE (pk1, pk2, ck1, ck2, ck3, v) VALUES (?, ?, ?, ?, ?, ?)",
+                vec![Int(0), Int(0), Int(0), Int(0), Int(0), Int(0)],
+            ),
+            (
+                "INSERT INTO RANGE_DELETE (pk1, pk2, ck1, ck2, ck3, v) VALUES (?, ?, ?, ?, ?, ?)",
+                vec![Int(0), Int(0), Int(1), Int(1), Int(1), Int(1)],
+            ),
+            (
+                "INSERT INTO RANGE_DELETE (pk1, pk2, ck1, ck2, ck3, v) VALUES (?, ?, ?, ?, ?, ?)",
+                vec![Int(0), Int(0), Int(2), Int(2), Int(2), Int(2)],
+            ),
+            (
+                "INSERT INTO RANGE_DELETE (pk1, pk2, ck1, ck2, ck3, v) VALUES (?, ?, ?, ?, ?, ?)",
+                vec![Int(0), Int(0), Int(3), Int(3), Int(3), Int(3)],
+            ),
+            (
+                "DELETE FROM RANGE_DELETE WHERE pk1 = ? AND pk2 = ? AND ck1 = ? AND ck2 > ? AND ck2 < ?",
+                vec![Int(0), Int(0), Int(0), Int(-1), Int(1)],
+            ),
+            (
+                "DELETE FROM RANGE_DELETE WHERE pk1 = ? AND pk2 = ? AND ck1 = ? AND ck2 < ?",
+                vec![Int(0), Int(0), Int(1), Int(2)],
+            ),
+            (
+                "DELETE FROM RANGE_DELETE WHERE pk1 = ? AND pk2 = ? AND (ck1, ck2) < ? AND (ck1, ck2, ck3) > ?",
+                vec![Int(0), Int(0), Tuple(vec![Some(Int(3)), Some(Int(3))]), Tuple(vec![Some(Int(2)), Some(Int(2)), Some(Int(2))])]
+            ),
```

Comment:
I added the following operation and your code fails:

```rust
(
    "DELETE FROM RANGE_DELETE WHERE pk1 = ? AND pk2 = ? AND (ck1, ck2, ck3) > ?",
    vec![Int(0), Int(0), Tuple(vec![Some(Int(2)), Some(Int(2)), Some(Int(2))])],
),
```

```
---- replication_tests::tests::test_range_delete stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Database returned an error: The submitted query has a syntax error, Error message: line 1:92 no viable alternative at input '('', scylla-cdc-replicator/src/replication_tests.rs:1346:14
```

I verified that this error comes from the replicator.